src/path.rs

```diff
@@ -0,0 +1,790 @@
+/// let runtime_dir_path = runtime_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn runtime_dir() -> crate::Result<PathBuf> {
+    let raw = inner::runtimeDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the path to the user's template directory.
+///
+/// #### Platform-specific
+///
+/// - **Linux:** Resolves to [`xdg-user-dirs`](https://www.freedesktop.org/wiki/Software/xdg-user-dirs/)' `XDG_TEMPLATES_DIR`.
+/// - **macOS:** Not supported.
+/// - **Windows:** Resolves to `{FOLDERID_Templates}`.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::template_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let template_dir_path = template_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn template_dir() -> crate::Result<PathBuf> {
+    let raw = inner::templateDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the path to the user's video directory.
+///
+/// #### Platform-specific
+///
+/// - **Linux:** Resolves to [`xdg-user-dirs`](https://www.freedesktop.org/wiki/Software/xdg-user-dirs/)' `XDG_VIDEOS_DIR`.
+/// - **macOS:** Resolves to `$HOME/Movies`.
+/// - **Windows:** Resolves to `{FOLDERID_Videos}`.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::video_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let video_dir_path = video_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn video_dir() -> crate::Result<PathBuf> {
+    let raw = inner::videoDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the path to the suggested directory for your app's log files.
+///
+/// #### Platform-specific
+///
+/// - **Linux:** Resolves to `${configDir}/${bundleIdentifier}/logs`.
+/// - **macOS:** Resolves to `${homeDir}/Library/Logs/{bundleIdentifier}`
+/// - **Windows:** Resolves to `${configDir}/${bundleIdentifier}/logs`.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::app_log_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_log_dir_path = app_log_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn app_log_dir() -> crate::Result<PathBuf> {
+    let raw = inner::appLogDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Resolves a sequence of `paths` or `path` segments into an absolute path.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::{resolve, app_data_dir};
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_data_dir_path = app_data_dir().await?;
+///
+/// let path = resolve([app_data_dir_path, "..", "users", "tauri", "avatar.png"]).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn resolve(paths: impl IntoIterator<Item = &str>) -> crate::Result<PathBuf> {
```

Comment:
Should this accept `AsRef<Path>` instead of `&str`?

Reply:
Updated in [a66b598a](https://github.com/emirror-de/tauri-sys/commit/a66b598a7a456b301290f8de5a5b37f5ab108813)